### PR TITLE
refactor(e2e): change fetch gateway retry mechanism

### PR DESCRIPTION
**Pull request by a-cordier**
---
### 🧪 End-to-End Coverage

| INSTRUCTIONS | BRANCHES |
| :----------: | :------: |
|   39% | 22%   |

A more detailed report has been uploaded to [circleci](https://output.circle-artifacts.com/output/job/9a65f4f4-aedc-458f-940c-c3b31524d888/artifacts/0/gravitee-apim-e2e/jacoco/reports/index.html)

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/test-retry-run-e2e/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-kedebzzieb.chromatic.com)

### CODE REVIEW

File: gravitee-apim-e2e/api-test/src/gateway/redeploy-api.spec.ts [hunk 1]
```diff
@@ -108,8 +108,6 @@ describe('Redeploy Api', () => {
       // Execute a request during which we will redeploy the api (only one try, no delay before call)
       const fetchUniqueGatewayCall = fetchGatewaySuccess({
         contextPath: createdApi.context_path + '?activeLatency=true',
-        failAfterMs: 0,
```
Comment (ThibaudAV) [inline gravitee-apim-e2e/api-test/src/gateway/redeploy-api.spec.ts:111]: maxRetries. because here we dont't want retry

Comment (ThibaudAV) [reply gravitee-apim-e2e/api-test/src/gateway/redeploy-api.spec.ts:111]: no ? 🤔

File: gravitee-apim-e2e/api-test/use-case-test/src/management/publisher/debug-mode/call-my-api-including-query-params-headers-body-and-view-debug-session-with-proper-info.spec.ts [hunk 2]
```diff
@@ -230,6 +231,10 @@ describe('Call my API (incl. query params, Headers and body) and view debug sess
       });
     });
 
+    afterEach(() => {
+      debugEventSubscription.unsubscribe();
```
Comment (ThibaudAV) [inline gravitee-apim-e2e/api-test/use-case-test/src/management/publisher/debug-mode/call-my-api-including-query-params-headers-body-and-view-debug-session-with-proper-info.spec.ts:235]: 👍  😁

File: gravitee-apim-e2e/api-test/use-case-test/src/management/publisher/flows/add-policies-to-flows-and-use-them.spec.ts [hunk 3]
```diff
@@ -198,8 +198,7 @@ describe('Add policies to flows and use them', () => {
         headers: {
           'x-cache-key': 'gravitee-test',
         },
-        timeout: 6000, // wait for cache eviction before calling the entrypoint
-      }).then((res) => res.json());
```
Comment (ThibaudAV) [inline gravitee-apim-e2e/api-test/use-case-test/src/management/publisher/flows/add-policies-to-flows-and-use-them.spec.ts:202]: 🤔  wanted. to remove the .json() ?

Comment (a-cordier) [reply gravitee-apim-e2e/api-test/use-case-test/src/management/publisher/flows/add-policies-to-flows-and-use-them.spec.ts:202]: yes, as there is no more timeout before calling, the assertion is moved to the response validator

File: gravitee-apim-e2e/api-test/src/gateway/redeploy-api.spec.ts (hunk 1 above)
Comment (a-cordier) [reply gravitee-apim-e2e/api-test/src/gateway/redeploy-api.spec.ts:111]: 0 does not work with this mechanism, but one does I'll wait for the current build to see how it goes and update after that

**Review by ThibaudAV — APPROVED** [review]

**Pull request merged**